bpf/bpf_sock.c

```diff
@@ -716,123 +673,38 @@ static __always_inline int __sock6_xlate(struct bpf_sock_addr *ctx)
 		return -ENOENT;
 	}
 
-	if (ctx->protocol != IPPROTO_TCP &&
-	    sock6_update_revnat(ctx, backend, &key,
+	if (!udp_only && ctx->protocol == IPPROTO_TCP) {
+		goto update_dst;
```

Comment:
Same comment as for `__sock4_xlate` applies here regarding `goto` and `update_dst` jump label with `__maybe_unused`.